docs/rpc-spec.md

```diff
@@ -1,8 +1,19 @@
+> [!IMPORTANT]
+> Since `4.1.0` (`rpc_version` 18), Transmission has switched from a bespoke RPC protocol to [JSON-RPC 2.0](https://www.jsonrpc.org/specification).
+> All RPC strings has been converted to snake_case as well.
+>
+> For backward compatibility, the old RPC protocol will still be available until `5.0.0`.
+> The old RPC strings consisting of a mix of kebab-case and camelCase are still available,
+> but they are deprecated and will be removed in `5.0.0` as well.
+>
+> For documentation of the old RPC protocol and strings, please consult documentation from previous versions.
+> https://github.com/transmission/transmission/blob/4.0.6/docs/rpc-spec.md
```

Comment:
I like the idea of deprecating & nudging downstream devs to using JSON-RPC, but I don't see any upside into committing to remove it in 5.0.0?

WDYT about something like

> Transmisson 4.1.0 added support for the [JSON-RPC 2.0](https://www.jsonrpc.org/specification) protocol and converted all RPC strings to snake_case.
>
> The old bespoke RPC protocol, and the old mix of kebab-case and camelCase strings, are still supported in Transmission 4 but are deprecated and will be removed in the future. People using the old protocol should update their code!
>
> For documentation of the old RPC protocol and strings, please consult documentation from previous versions.
> https://github.com/transmission/transmission/blob/4.0.6/docs/rpc-spec.md

Reply:
Sure, I don't mind leaving the official announcement undefined.

I will make the change over at #7269 for now, and once that is merged I will rebase and update this PR as well.